core/src/channel_group/mod.rs

```diff
@@ -16,15 +16,35 @@ const MAX_EVENT_CACHE_SIZE: u32 = 1024 * 1024;
 ///
 /// Manages multiple VoiceChannel objects at once.
 pub struct ChannelGroup {
-    thread_pool: rayon::ThreadPool,
+    thread_pool: Option<rayon::ThreadPool>,
     cached_event_count: u32,
     channel_events_cache: Box<[Vec<ChannelAudioEvent>]>,
     sample_cache_vecs: Box<[Vec<f32>]>,
     channels: Box<[VoiceChannel]>,
     audio_params: AudioStreamParams,
 }
 
+/// Options regarding which parts of the ChannelGroup should be multithreaded.
+///
+/// The following apply for all the values:
+/// - A value of `None` means no multithreading.
+/// - If the value is set to `Some(0)` then the number of threads will be
+///     determined automatically by `rayon`. Please read
+///     [this](https://docs.rs/rayon-core/1.11.0/rayon_core/struct.ThreadPoolBuilder.html#method.num_threads)
+///     for more information.
+#[derive(Clone)]
+pub struct ParallelismOptions {
```

Comment:
> E.g. "pub enum ThreadCount" with options One, Auto, Manual(u32) or something like that, where Auto just uses the core count

True that sounds like a better approach. Should also be easier for the user to understand what is going on.

> Also It's nice to have constants for these things, because 90% of use cases are just "auto multithreading per channel" or "auto multithreading per channel per key"

You mean make a constant of the enum:
```rust
const AUTO_MULTITHREADING: ThreadCount = ThreadCount::Auto
```
or the struct itself:
```rust
const AUTO_MULTITHREADING: ParallelismOptions = ParallelismOptions {
    channel: ThreadCount::Auto,
    key: ThreadCount::Auto,
}
```
?

Reply:
Nah I meant making it a constant implemented on the struct, you can have `const` fields inside `impl` blocks

Reply:
Yeah that, except AUTO_PER_KEY (key and channel) and AUTO_PER_CHANNEL (just channel) with descriptions for each, and an Default implementation that does auto per key

Reply:
In my opinion at least

Reply:
> Yeah that, except AUTO_PER_KEY (key and channel) and AUTO_PER_CHANNEL (just channel) with descriptions for each, and an Default implementation that does auto per key

Alright yeah that sounds good. Shouldn't default be auto for both though? It was like that before (at least for channel).